### PR TITLE
test: Fix pod quota test for kubernetes

### LINKED ISSUE #1513: fix pod quota test for kubernetes
Fix pod quota test to avoid leaving pod information at /var/lib/vc/sbs and verify after running the tests that not pod information is left

### PR DESCRIPTION
**Pull request by GabyCT**
Fix pod quota test to avoid leaving pod information at /var/lib/vc/sbs (issue: https://github.com/kata-containers/runtime/issues/1611) and verify after running the tests that not pod information is left.

Fixes #1513

Signed-off-by: Gabriela Cervantes <gabriela.cervantes.tellez@intel.com>

### CODE REVIEW

File: integration/kubernetes/runtimeclass_workloads/pod-quota-deployment.yaml [hunk 1]
```diff
@@ -11,7 +11,7 @@ spec:
   selector:
     matchLabels:
       purpose: quota-demo
-  replicas: 3
+  replicas: 2
```
Comment (grahamwhaley) [inline integration/kubernetes/runtimeclass_workloads/pod-quota-deployment.yaml]: why this change? Is it related to not leaving pods around?

File: integration/kubernetes/cleanup_env.sh [hunk 2]
```diff
@@ -35,3 +35,6 @@ sudo ip link del flannel.1
 
 # Check no kata processes are left behind after reseting kubernetes
 check_processes
+
+# Checks that pods were not left
+check_pods
```
Comment (grahamwhaley) [inline integration/kubernetes/cleanup_env.sh]: nice - I presume this will not check on all k8s tests we run?

File: integration/kubernetes/runtimeclass_workloads/pod-quota-deployment.yaml (hunk 1 above)
Comment (GabyCT) [reply integration/kubernetes/runtimeclass_workloads/pod-quota-deployment.yaml]: @grahamwhaley yes, with this we will not leave pods

File: integration/kubernetes/cleanup_env.sh (hunk 2 above)
Comment (GabyCT) [reply integration/kubernetes/cleanup_env.sh]: @grahamwhaley yes it will

Comment (grahamwhaley) [reply integration/kubernetes/cleanup_env.sh]: Oops, I meant 'will check' - good !

File: integration/kubernetes/runtimeclass_workloads/pod-quota-deployment.yaml (hunk 1 above)
Comment (grahamwhaley) [reply integration/kubernetes/runtimeclass_workloads/pod-quota-deployment.yaml]: but... why? Can you explain (maybe in the commit message) why 3->2 makes it not leak?

Comment (chavafg) [reply integration/kubernetes/runtimeclass_workloads/pod-quota-deployment.yaml]: @GabyCT can you open an issue in the runtime repo and add it as a reference maybe in the commit message?

Comment (GabyCT) [reply integration/kubernetes/runtimeclass_workloads/pod-quota-deployment.yaml]: @chavafg and @grahamwhaley changes applied

Comment (jodh-intel) [inline integration/kubernetes/runtimeclass_workloads/pod-quota-deployment.yaml]: Like @grahamwhaley, I'm still not clear on this change. It might "work" with `2`, but why did we choose `3` initially then? What did we break?

Maybe @egernst or @mcastelino have thoughts?

Comment (mcastelino) [reply integration/kubernetes/runtimeclass_workloads/pod-quota-deployment.yaml]: @GabyCT there does not seem to be a co-relation between the bug and this fix. Can you expand? Were we failing deployment initially?

Comment (GabyCT) [reply integration/kubernetes/runtimeclass_workloads/pod-quota-deployment.yaml]: @jodh-intel and @mcastelino before this PR, we never check if kubernetes tests were leaving pod information behind. Until recently, we found out that this test was leaving pod information that is why as part of this PR, I am including to check that after running all the kubernetes tests. If we leave it like it is with 3 replicas then we will have pod info at /var/lib/vc/sbs. The deployment of the replicas is done successfully and even it says that when we remove the deployment is donde successfully, however, after cleaning everything the pod info is at /var/lib/vc/sbs.

Comment (GabyCT) [reply integration/kubernetes/runtimeclass_workloads/pod-quota-deployment.yaml]: @jodh-intel , we choose 3 before because the test was trying to run 3 containers but the resource quota was only 2 so the test verified that it was not possible to run 3 containers and that only 2 were running
